src/common.ts

```diff
@@ -33,8 +33,28 @@ export function processProperty(prop: Schema, name = '', namespace = '',
 
     const list = prop.enum || prop.items.enum;
     const exp = exportEnums ? 'export ' : '';
-    enumDeclaration = `${exp}type ${type} =\n` + indent('\'' + list.join('\' |\n\'')) + '\';';
-
+    console.log(prop.type);
+
+    const generateEnum = (originType: string, exp: string, type: string, list: any[], $ref: string): string => {
+      console.log(originType);
+      switch(originType) {
+        case undefined:
+          const defType = translateType($ref);
+          return generateEnum(defType.type, exp, type, list, undefined);
+        case 'integer':
+          return `${exp}type ${type} =\n` + indent(list.join('|\n')) + ';';
+        case 'string':
+          return `${exp}type ${type} =\n` + indent('\'' + list.join('\' |\n\'')) + '\';';
+        default:
+        if (typeof list[0] === 'number') {
+          return `${exp}type ${type} =\n` + indent(list.join('|\n')) + ';';
+        } else {
+          return `${exp}type ${type} =\n` + indent('\'' + list.join('\' |\n\'')) + '\';';
+        }
+      }
+    } 
+    enumDeclaration = generateEnum(prop.type, exp, type, list, prop.$ref);
```

Comment:
Anyway, I would imagine a bit simpler modification that takes care of (diverts) inner `list.join` only. Please, try to simplify.